app/controllers/media_objects_controller.rb

```diff
@@ -334,7 +334,9 @@ def show
         end
       end
       format.json do
-        render json: @media_object.to_json
+        response_json = @media_object.to_json
+        response_json = JSON.parse(response_json).except('files', 'visibility', 'read_groups').to_json unless current_ability.is_administrator?
```

Comment:
I think we can avoid the JSON parse by using `as_json` which should return the hash that gets used to create the JSON string:
```
response_json = @media_object.as_json
response_json = response_json.except('files', 'visibility', 'read_groups') unless current_ability.is_administrator?
render json: response_json.to_json
```